aptos-move/e2e-tests/src/executor.rs

```diff
@@ -1063,51 +1071,45 @@ impl FakeExecutor {
                 self.env.clone(),
                 &resolver,
             );
+            let module_storage = self.data_store.as_aptos_code_storage(&vm);
             let mut session = vm.new_session(&resolver, SessionId::void(), None);
             let storage = TraversalStorage::new();
             session
                 .execute_function_bypass_visibility(
-                    module_id,
+                    &module_id,
                     &Self::name(function_name),
                     type_params,
                     args,
                     // TODO(Gas): we probably want to switch to metered execution in the future
                     &mut UnmeteredGasMeter,
                     &mut TraversalContext::new(&storage),
-                    &DummyCodeStorage,
+                    &module_storage,
                 )
                 .unwrap_or_else(|e| {
                     panic!(
                         "Error calling {}.{}: {}",
-                        module_id,
+                        &module_id,
                         function_name,
                         e.into_vm_status()
                     )
                 });
-            let (change_set, module_write_set) = session
-                .finish(&ChangeSetConfigs::unlimited_at_gas_feature_version(
-                    LATEST_GAS_FEATURE_VERSION,
-                ))
+            let (change_set, empty_module_write_set) = session
```

Comment:
nit: can we refactor this out together with (it seems the exact same thing) above 1034-1045, and maybe even elsewhere (e.g. simulation) into a easy utility method that asserts empty module write set and produces change set from session

Reply:
done, [aptos-move/e2e-tests/src/executor.rs](https://github.com/aptos-labs/aptos-core/pull/14206/files/f7a4c4de8ad773129342209b845f20ee0cce9e8a#diff-9189ada6802b4ca39f97b636f9745e595c3681ddd5f2eda6ea2e33adc3445565) definitely needs some more love